integrationtests/foldersync/foldersync_test.go

```diff
@@ -0,0 +1,385 @@
+
+	db3, err := db.NewDBFromAddr(context.Background(), network3, creds, addrs0[0], key0, db.WithRepoPath(repoPath3), db.WithCollections(cc))
+	checkErr(t, err)
+	defer db3.Close()
+
+	c3 := db3.GetCollection(collectionName)
+
+	// Add some data
+
+	folder0 := folder{ID: core.NewInstanceID(), Owner: "client0", Files: []file{}}
+	folder1 := folder{ID: core.NewInstanceID(), Owner: "client1", Files: []file{}}
+	folder2 := folder{ID: core.NewInstanceID(), Owner: "client2", Files: []file{}}
+	folder3 := folder{ID: core.NewInstanceID(), Owner: "client3", Files: []file{}}
+
+	_, err = c0.Create(util.JSONFromInstance(folder0))
+	checkErr(t, err)
+	_, err = c1.Create(util.JSONFromInstance(folder1))
+	checkErr(t, err)
+	_, err = c2.Create(util.JSONFromInstance(folder2))
+	checkErr(t, err)
+	_, err = c3.Create(util.JSONFromInstance(folder3))
+	checkErr(t, err)
+
+	time.Sleep(time.Second * 15)
+
+	instances0, err := c0.Find(nil)
+	checkErr(t, err)
+	if len(instances0) != 4 {
+		t.Fatalf("expected 4 instances, but got %v", len(instances0))
+	}
+
+	instances1, err := c1.Find(nil)
+	checkErr(t, err)
+	if len(instances1) != 4 {
+		t.Fatalf("expected 4 instances, but got %v", len(instances1))
+	}
+
+	instances2, err := c2.Find(nil)
+	checkErr(t, err)
+	if len(instances2) != 4 {
+		t.Fatalf("expected 4 instances, but got %v", len(instances2))
+	}
+
+	instances3, err := c3.Find(nil)
+	checkErr(t, err)
+	if len(instances3) != 4 {
+		t.Fatalf("expected 4 instances, but got %v", len(instances3))
+	}
+}
+
+func TestNUsersBootstrap(t *testing.T) {
+	tests := []struct {
+		totalClients     int
+		totalCorePeers   int
+		syncTimeout      time.Duration
+		randFilesGen     int
+		randFileSize     int
+		checkSyncedFiles bool
+	}{
+		// {totalClients: 2, totalCorePeers: 1, syncTimeout: time.Second * 5},
+		{totalClients: 5, totalCorePeers: 1, syncTimeout: time.Second * 15},
+
+		// {totalClients: 5, totalCorePeers: 2, syncTimeout: time.Second * 20},
+
+		// {totalClients: 2, totalCorePeers: 1, syncTimeout: time.Second * 10, randFilesGen: 4, randFileSize: 10},
+		// {totalClients: 5, totalCorePeers: 4, syncTimeout: time.Second * 20, randFilesGen: 4, randFileSize: 10},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(fmt.Sprintf("Total%dCore%d", tt.totalClients, tt.totalCorePeers), func(t *testing.T) {
+			t.Parallel()
+			var clients []*client
+
+			client0, clean0 := createRootClient(t, fmt.Sprintf("client%d", 0))
+			defer clean0()
+			clients = append(clients, client0)
+
+			tid, addr, key, err := client0.getInviteInfo()
+			checkErr(t, err)
+
+			for i := 1; i < tt.totalCorePeers; i++ {
+				client, clean := createJoinerClient(t, fmt.Sprintf("client%d", i), tid, addr, key)
+				defer clean()
+				clients = append(clients, client)
+			}
+
+			for i := tt.totalCorePeers; i < tt.totalClients; i++ {
+				index := i % tt.totalCorePeers
+				fmt.Printf("using index %v for client %v\n", index, i)
+				tid, addr, key, err := clients[i%tt.totalCorePeers].getInviteInfo()
```

Comment:
Not sure who originally wrote this part, but can @sanderpick or @jsign explain what the intention of non-core peers is where this modulo is used to choose the peer to get the invite from? Thing is, in all these tests, that always evaluates to 0.

Reply:
That's because I think someone commented other tests setups where `totalCorePeers` are greater than 1. If you uncomment those, it evaluates to other values than 0.

The idea is to make peers connect to `totalCorePeers` in invitations. Cycling thought them.

Reply:
Sorry, I mis-spoke... there is one instance where it evaluates to 1, but the rest of the time it's 0.

Reply:
@asutula, I think we both commented at the same time.
See above just in case you missed it.

Reply:
To be clear, that `i%tt.totalCorePeers` is just selecting a core peer for some non-core client. If its 0, or 1, or whatever shouldn't be important. 

It's just distributing invitations evenly on core-peers, that's the intention of the code.

Reply:
Ok thanks for the explanation. My statement about it evaluating to 1 in a single case is with all the tests enabled here locally. Since it was non-0 in only that one instance, it made me question if something was wrong, but I get what you're saying... it's a general rule that is applied to every test no matter what the test configuration is.

Reply:
Exactly!